RecoverCRPVM/RecoverCRPVM.ps1

```diff
@@ -0,0 +1,175 @@
+	-Detach the OS disk that was attached as a data disk to the rescue VM
+	-Disk Swap  OS Disk of the Problem VM with the fixed OS Disk Uri
+    -Start the VM
+    -Delete all the resources that were created for the rescue VM
+
+.PARAMETER VmName
+    This is a mandatory Parameter, Name of the VM that needs to be recovered.
+
+.PARAMETER ResourceGroup
+    This is a mandatory Parameter, Name of the ResourceGroup the VM belongs to.
+
+.PARAMETER SubID
+    This is a mandatory Parameter, SubscriptionID - the VM belongs to.
+
+.PARAMETER FixedOsDiskUri
+    This is a mandatory Parameter, This would be the uri of the fixedOSDisk, this information will be provided after the successful execution of CreateCRPRescueVM.
+
+.PARAMETER prefix
+    Optional Parameter. By default the new Rescue VM and its resources are all created under a ResourceGroup named same as the orginal resourceGroup name with a prefix of 'rescue', however the prefix can be changed to a different value to overide the default 'resuce'
+
+
+
+.EXAMPLE
+    .\RecoverCRPVM.ps1 -ResourceGroup "rescueportalLin" -VmName "ubuntu2" -SubID "xxxxxxxx-abdf-4aaf-8868-2002dfeea60c" -FixedOsDiskUri "https://vmrecoverytestdisks645.blob.core.windows.net/vhds/fixedosfixedosubuntu220171220164151.vhd" -prefix "rescuered"
+
+.NOTES
+    Name: CreateCRPRescueVM.ps1
+
+    Author: Sujasd
+#>
+
+Param(
+        [Parameter(mandatory=$true)]
+        [String]$VmName,
+
+        [Parameter(mandatory=$true)]
+        [String]$ResourceGroup,
+
+        [Parameter(mandatory=$true)]
+        [String]$SubID,
+
+        [Parameter(mandatory=$true)]
+        [String]$FixedOsDiskUri,
+
+        [Parameter(mandatory=$false)]
+        [String]$prefix = "rescue"
+     )
+
+
+$Error.Clear()
+cls
+if (-not $showErrors) {
+    $ErrorActionPreference = 'SilentlyContinue'
+}
+
+$script:scriptStartTime = (Get-Date).ToUniversalTime()
+$LogFile = $env:TEMP + "\DiskSwap" + $VmName + "_" + ( Get-Date $script:scriptStartTime -f yyyyMMddHHmmss ) + ".log"  
+# Get running path
+$RunPath = split-path -parent $MyInvocation.MyCommand.Source
+cd $RunPath
+$CommonFunctions = $runPath+"\Common-Functions.psm1"
+
+$CommonFunctions = $runPath+"\Common-Functions.psm1"
+
+if (Get-Module Common-Functions) {remove-module -name Common-Functions}   
+Import-Module -Name $CommonFunctions  -ArgumentList $LogFile -ErrorAction Stop 
+
+if (-not (Get-AzureRmContext).Account)
+{
+    Login-AzureRmAccount
+}
+write-log "Info: Log is being written to ==> $LogFile" 
+#Set the context to the correct subid
+Write-Log "Setting the context to SubID $SubID" -color Yellow
+$subContext= Set-AzureRmContext -Subscription $SubID
+if ($subContext -eq $null) 
+{
+    Write-Log "Unable to set the Context for the given subId ==> $SubID, Please make sure you first  run the command ==> Login-AzureRMAccount" -Color Red
+    return
+}
+
+#Step 1 Get the VM Object
+Write-Log "Running Get-AzureRmVM -ResourceGroupName `"$ResourceGroup`" -Name `"$VmName`"" -Color Yellow
```

Comment:
Lo-pri: Unless it slows down the process, it would be good to have an extra validation that $RescueResourceGroup exists